decode.go

```diff
@@ -96,7 +96,11 @@ type Decoded struct {
 	pos         int64
 }
 
-func (d *Decoded) readFrame() error {
+// Decoded is the old type name for the Decoder
+// DEPRECATED
+type Decoded Decoder
```

Comment:
I'd define this as a struct with a non-named member Decoder so that we can use same methods as Decoder